src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -19,33 +26,75 @@ public class PullRequestService : IPullRequestService
+        readonly IGitClient gitClient;
+        readonly IGitService gitService;
+        readonly IOperatingSystem os;
+
+        [ImportingConstructor]
+        public PullRequestService(IGitClient gitClient, IGitService gitService, IOperatingSystem os)
+        {
+            this.gitClient = gitClient;
+            this.gitService = gitService;
+            this.os = os;
+        }
+
+        public IObservable<IPullRequestModel> CreatePullRequest(IRepositoryHost host,
+            ISimpleRepositoryModel sourceRepository, ISimpleRepositoryModel targetRepository,
+            IBranch sourceBranch, IBranch targetBranch,
+            string title, string body
+        )
         {
             Extensions.Guard.ArgumentNotNull(host, nameof(host));
-            Extensions.Guard.ArgumentNotNull(repository, nameof(repository));
+            Extensions.Guard.ArgumentNotNull(sourceRepository, nameof(sourceRepository));
+            Extensions.Guard.ArgumentNotNull(targetRepository, nameof(targetRepository));
+            Extensions.Guard.ArgumentNotNull(sourceBranch, nameof(sourceBranch));
+            Extensions.Guard.ArgumentNotNull(targetBranch, nameof(targetBranch));
             Extensions.Guard.ArgumentNotNull(title, nameof(title));
             Extensions.Guard.ArgumentNotNull(body, nameof(body));
-            Extensions.Guard.ArgumentNotNull(source, nameof(source));
-            Extensions.Guard.ArgumentNotNull(target, nameof(target));
 
-            return host.ModelService.CreatePullRequest(repository, title, body, source, target);
+            return PushAndCreatePR(host, sourceRepository, targetRepository, sourceBranch, targetBranch, title, body).ToObservable();
         }
 
-        public string GetPullRequestTemplate(ISimpleRepositoryModel repository)
+        public IObservable<string> GetPullRequestTemplate(ISimpleRepositoryModel repository)
         {
             Extensions.Guard.ArgumentNotNull(repository, nameof(repository));
 
-            var paths = TemplatePaths.Select(x => Path.Combine(repository.LocalPath, x));
-
-            foreach (var path in paths)
+            return Observable.Defer(() =>
             {
-                if (File.Exists(path))
+                string ret = null;
+                var paths = TemplatePaths.Select(x => Path.Combine(repository.LocalPath, x));
+
+                foreach (var path in paths)
                 {
-                    try { return File.ReadAllText(path); } catch { }
+                    if (os.File.Exists(path))
+                    {
+                        try { ret = os.File.ReadAllText(path, Encoding.UTF8); } catch { }
+                        if (ret != null)
+                            break;
+                    }
                 }
-            }
+                return Observable.Return(ret);
```

Comment:
This will return an observable that contains `null` if a template is not found, but it should probably return an empty sequence. It was returning `null` before to say "template file not found" but now we're using observables an empty sequence is a more natural fit for that,